awx_collection/test/awx/conftest.py

```diff
@@ -209,3 +209,19 @@ def vault_credential(organization):
         credential_type=ct, name='vault-cred',
         inputs={'vault_id': 'foo', 'vault_password': 'pas4word'}
     )
+
+
+@pytest.fixture
+def silence_deprecation():
+    """The deprecation warnings are stored in a global variable
+    they will create cross-test interference. Use this to turn them off.
+    """
+    with mock.patch('ansible.module_utils.basic.AnsibleModule.deprecate'):
+        yield
```

Comment:
wondering if it might be good to assert that the method was called at least once, so that unnecessary uses of this fixture don't hang around. But that wouldn't be workable if different Ansible versions behaved differently, so actually that might not work.